.github/actions/cache-pip/action.yml

```diff
@@ -5,7 +5,7 @@ runs:
   steps:
     - uses: ./.github/actions/py-cache-key
       id: py-cache-key
-    - uses: actions/cache@v4
+    - uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
```

Comment:
If an attacker obtained permission to modify the tag, they could execute the following steps to run malicious code and exfiltrate secrets:

1. Fork the [`actions/cache`](https://github.com/actions/cache) repository.  
2. Create a commit containing malicious code.  
3. Repoint the `v4` tag to the malicious commit in the fork.

a full length commit SHA can prevent that.

Reply:
This attack was used in the following supply chain attack:

https://unit42.paloaltonetworks.com/github-actions-supply-chain-attack/